src/components/UI/Cards/HistoyryVideoCards/HistoryVideoCards.jsx

```diff
@@ -0,0 +1,68 @@
+import axios from 'axios';
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { useHistory } from '../../../../context/HistoryContext';
+import ViewCount from '../../../../utils/ViewCount';
+import IcRoundAutoModeAddWatchLater from '../../Icons/IcRoundAutoMode';
+import IcRoundCheckCircleConfirmWatchLater from '../../Icons/IcRoundCheckCircle';
+import IcTwotoneDelete from '../../Icons/IcTwotoneDelete';
+import "./HistoryVideoCards.css";
+
+function HistoryVideoCards({ props}) {
+    const { historyContextArray, setHistoryContextArray } = useHistory();
+    const { videoid, snippet, statistics } = props;
+    console.log(props, videoid, snippet, statistics)
+    const deleteHistoryHandler = async(id) => { 
+        try {
+            console.log(id)
+            var res = await axios.delete(`/api/user/history/${id}`,{ headers:{
+                authorization:localStorage.getItem("jafnaToken")}});
+            console.log(res);
+            // setHistoryContextArray();
+            (async() => {
+                var res = await axios.get("/api/user/history",{ headers:{
+                  authorization:localStorage.getItem("jafnaToken")}});
+                console.log(res);
+                // console.log(res.data.products);
+                setHistoryContextArray(res.data.history);
+               
+            })()
+        }
```

Comment:
What are you doing here? 

In your first call you will get response and you can use that response to set the item instead of doing another async all

Reply:
I am not sure about the response it returns if after deleting the specific history it is returning the filtered i will set that 
Thanks for bringing this up